src/scenarios/shared/mauisharedpython.py

```diff
@@ -1,24 +1,95 @@
+        raise ValueError("Unable to find general version in Version.Details.xml")
+
+    # Get the available versions from the Version.Details.xml file
+    dependencies = root.findall(".//Dependency[@Name]")
+    for rollback_name, xml_name in rollback_name_to_xml_name_mappings.items():
+        for dependency in dependencies:
+            if dependency.get("Name").startswith(xml_name): # type: ignore we know Name is present
+                workload_version = dependency.get("Version", "ERROR: Failed to get version")
+                if workload_version == "ERROR: Failed to get version":
+                    raise ValueError(f"Unable to find {xml_name} with proper version in the provided xml file")
+
+                # Use the band version based on what the maui upstream currently has. This is necessary if they hardcode the version.
+                band_name_match_string = rf"^\s*Mapping_{xml_name}:(\S*)"
+                band_version_mapping = re.search(band_name_match_string, version_details_xml, flags=re.MULTILINE)
+                if band_version_mapping is None:
+                    raise ValueError(f"Unable to find band version mapping for match {band_name_match_string} in Version.Details.xml")
+                if band_version_mapping.group(1) == "default":
+                    band_version = default_band_version
+                else:
+                    band_version = band_version_mapping.group(1)
+                rollback_dict[rollback_name] = f"{workload_version}/{band_version}"
+                break
+        if rollback_name not in rollback_dict:
+            raise ValueError(f"Unable to find {rollback_name} with proper version in Version.Details.xml")
+    return rollback_dict
+
+def dump_dict_to_json_file(dump_dict: dict[str, str], file_name: str):
+    json_output = json.dumps(dump_dict, indent=4)
+    with open(file_name, "w", encoding="utf-8") as f:
+        f.write(json_output)
 
 def install_versioned_maui(precommands: PreCommands):
     target_framework_wo_platform = precommands.framework.split('-')[0]
 
     # Download what we need
-    with open ("MauiNuGet.config", "wb") as f:
-        f.write(requests.get(f'https://raw.githubusercontent.com/dotnet/maui/{target_framework_wo_platform}/NuGet.config', allow_redirects=True).content)
+    with open("MauiNuGet.config", "wb") as f:
+        with urllib.request.urlopen(f'https://raw.githubusercontent.com/dotnet/maui/{target_framework_wo_platform}/NuGet.config') as response:
+            f.write(response.read())
 
     workload_install_args = ['--configfile', 'MauiNuGet.config', '--skip-sign-check']
-    if int(target_framework_wo_platform.split('.')[0][3:]) > 7: # Use the rollback file for versions greater than 7
-        workload_install_args += ['--from-rollback-file', f'https://maui.blob.core.windows.net/metadata/rollbacks/{target_framework_wo_platform}.json']
+    if int(target_framework_wo_platform.split('.')[0][3:]) > 8: # Use the rollback file for versions greater than 7
```

Comment:
Doesn't this use the rollback file for versions greater than 8?

Reply:
Yes, updated the number and left some information on what the number is for.